src/if.js

```diff
@@ -43,20 +43,42 @@ export class If {
   * @param newValue The new value
   */
   valueChanged(newValue) {
+    if (this.__queuedChanges) {      
+      this.__queuedChanges.push(newValue);
+      return;
+    }
+
+    let maybePromise = this._runValueChanged(newValue);
+    if (maybePromise instanceof Promise) {
+      let queuedChanges = this.__queuedChanges = [];
+
+      let runQueuedChanges = () => {
+        if (!queuedChanges.length) {
+          delete this.__queuedChanges;       
+          return;
+        }        
+        let nextPromise = this._runValueChanged(queuedChanges.shift()) || Promise.resolve();          
```

Comment:
@martingust, thanks for the reply.
> If a new change happens before an animation (ie the method valueChanged) has finished the new value will be pushed to __queuedChanges on line 47

That was clear to me,

however following
> `queuedChanges.length` will be `> 0`

is not clear to me, as it seems to me, that my previous statement:
> 1) `let queuedChanges = this.__queuedChanges = [];` always assigns empty array to `queuedChanges`

... **still holds even if `this.__queuedChanges` is not empty**. Let me create a similar setup (difference is that in this demo `__queuedChanges` is var instead of instance field):
```javascript
var __queuedChanges = [];
__queuedChanges.push({});

var queuedChanges = this.__queuedChanges = [];
console.log("queuedChanges.length =", queuedChanges.length);
```
produces `queuedChanges.length = 0`